### ISSUE TITLE
Submit button for search: add discernible text - Accessibility

### ISSUE THREAD

**Issue by lhenze** [inveniosoftware/invenio-app-rdm#1117]
Buttons must have discernible text, according to Accessibility guidelines.
https://www.w3.org/WAI/standards-guidelines/act/rules/button-non-empty-accessible-name-97a4e1/

For the search submit button that appears on the homepage and in the navbar 

Fix at least one (1) of these issues:
Element does not have inner text that is visible to screen readers

aria-label attribute does not exist or is empty

aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty

Element has no title attribute

Element's default semantics were not overridden with role="none" or role="presentation"
